model_api/cpp/models/src/segmentation_model.cpp

```diff
@@ -54,6 +68,34 @@ cv::Mat create_hard_prediction_from_soft_prediction(const cv::Mat& soft_predicti
     }
     return hard_prediction;
 }
+
+std::vector<Annotation> create_annotation_from_segmentation_map(const cv::Mat& hard_prediction, const cv::Mat& soft_prediction, std::vector<std::string> label_map) {
```

Comment:
```suggestion
std::vector<Annotation> create_annotation_from_segmentation_map(const cv::Mat& hard_prediction, const cv::Mat& soft_prediction, const std::vector<std::string>& label_map) {
```

Reply:
> How about having `create_annotation_from_segmentation_map()` as a stand alone function. So model wrapper produces outputs without contours, but if someone needs them, they can call `create_annotation_from_segmentation_map()` on the model output?

I do like the idea, but the returned object is a result object. The soft prediction is already lost at that point, which is required to generate the contours. It would make more sense to lock it behind a config in that case. WDYT?

Reply:
There's `return_soft_prediction` that makes the soft prediction available. The function generating contours should try to cast the returned object to `ImageResultWithSoftPrediction` and simply throw if the cast fails.

Reply:
Done.